### PR TITLE
Button with icon only - New Story

### LINKED ISSUE #52: Button with Icon only
We need to have the possibility to render a button without any text, only a picto :
Figma : 
https://www.figma.com/design/ZCSD09UXWl2lwFlb7nJqba/Mockups?node-id=257-28929&t=OF1m9lUKfxutn8yw-0

image

### PR DESCRIPTION
**Pull request by Rcaradec**
Close #52

### CODE REVIEW

**Comment by MatouTechnologies** [comment]
This pull request has been deployed to Vercel.
<table>
  <tr>
    <td><strong>✅ Preview:</strong></td>
    <td><a href='https://matou-technologies-purrfect-ui-storybook-89.vercel.app'>https://matou-technologies-purrfect-ui-storybook-89.vercel.app</a></td>
  </tr>
  <tr>
    <td><strong>🔍 Inspect:</strong></td>
    <td><a href='https://vercel.com/matou-technologies/purrfect-ui/F7US7AiFSSe3u1r5r8gqZDXZwew8'>https://vercel.com/matou-technologies/purrfect-ui/F7US7AiFSSe3u1r5r8gqZDXZwew8</a></td>
  </tr>
</table>


<hidden purpose="for-rewritable-pr-comment-action-use" value="deploy-preview-storybook"></hidden>

**Comment by MarcoPOtard** [comment]
I wonder if it would be interesting if the button was square and not rectangular ?

File: src/action/Button/Button.tsx [hunk 1]
```diff
@@ -39,77 +39,76 @@ const buttonVariant = (variant: Variant = "primary") => {
...
 export type Size = "small" | "normal" | "large";
-const buttonSize = (size: Size = "normal") => {
+const buttonSize = (size: Size = "normal", hasChildren: boolean = true) => {
   switch (size) {
     case "small":
       return css`
         gap: ${sizing(6)};
-        height: ${sizing(20)};
-        padding: 0px ${sizing(14)};
+        ${hasChildren && `padding: 0px ${sizing(14)};`}
```
Comment (MarcoPOtard) [inline src/action/Button/Button.tsx:47]: I think it's necessary to keep the padding for the button with children AND the button with only icon.
The icon button is naturally smaller than the button with text, but if you remove the padding when there is only an icon, the button is really small.
So I think is better to have padding fot icon button too

Comment (Rcaradec) [reply src/action/Button/Button.tsx:47]: good point, I'll fix it

**Review by MarcoPOtard — APPROVED** [review]

**Pull request merged**